xinference/model/multimodal/core.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2022-2023 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from typing import Tuple
+
+from pydantic import BaseModel
+
+from ...constants import XINFERENCE_CACHE_DIR
+from ..core import ModelDescription
+from .stable_diffusion.core import DiffusionModel
+
+MAX_ATTEMPTS = 3
+
+logger = logging.getLogger(__name__)
+
+
+class MultimodalModelSpec(BaseModel):
```

Comment:
In `xinference`, a "model family" refers to a group of models sharing similar characteristics such as network structure. Each model within this family has its own detailed specification, termed "model spec."

Given this, perhaps a more suitable name for this class would be `MultimodalModelFamilyV1`. The `V1` suffix aids in differentiating potential incompatible versions.

Reply:
Fixed. Thanks.